backends/ebpf/psa/README.md

```diff
@@ -11,9 +11,16 @@ This directory implements PSA (Portable Switch Architecture) for the eBPF backen
 
 # Design
 
+The PSA to eBPF compiler provides two flavors of generated eBPF code: TC-based design and XDP-based design.
+The TC-based design leverages eBPF TC (Traffic Control) hook and is able to implement any PSA program at the cost of performance. 
+The XDP-based design offloads packet processing to eBPF XDP (eXpress Data Path) hook and provides better performance than the TC-based flavor.
+However, the XDP-based design lacks support for packet recirculation, QoS (no integration with TC qdisc) and CLONE_E2E packet path.
+
+## TC-based design (default)
+
 P4 packet processing is translated into a set of eBPF programs attached to the TC hook. The eBPF programs implement packet processing defined
 in a P4 program written according to the PSA model. The TC hook is used as a main engine, because it enables a full implementation of the PSA specification.
-We also plan to contribute the XDP-based version of the PSA implementation that does not implement the full specification, but provides better performance.
+The XDP-based version of the PSA implementation does not implement the full specification, but provides better performance.
```

Comment:
can you provide a ballpark estimate of the improvement so people can decide which one to use?

Reply:
Yes, I think we could @tatry, but the improvement depends on the P4 program. A ball park estimate is over 2x improvement in throughput per CPU core measured in MPPS.

You can find an extensive performance evaluation in our recent paper: https://dl.acm.org/doi/10.1145/3555050.3569117. We could also reference this paper in the README?